unittests/test_versioning.py

```diff
@@ -0,0 +1,27 @@
+import unittest
+
+from reframe.utility.versioning import Version
+
+
+class TestVersioning(unittest.TestCase):
+    def test_version_format(self):
+        Version('1.2')
+        Version('1.2.3')
+        Version('1.2-dev0')
+        Version('1.2-dev5')
+        Version('1.2.3-dev2')
+        self.assertRaises(ValueError, Version, '1')
+        self.assertRaises(ValueError, Version, '1.2a')
+        self.assertRaises(ValueError, Version, 'a.b.c')
+        self.assertRaises(ValueError, Version, '1.2.3-dev')
```

Comment:
You should also check if sb calls you by passing `None` or an empty string.

Reply:
The unit test with `None` would through `AttributeError` but I don't know if `Version` should check if the version number that is passed is not an string. Because when once we have the `Field` implemented for the versioning, then Version will always get a string, right?

Reply:
`Version` should check for `None`s and disallow them raising a `ValueError`.